lib/modules/dosomething/dosomething_user/dosomething_user.module

```diff
@@ -465,6 +465,11 @@ function dosomething_user_new_user($form, &$form_state) {
   global $user;
   $account = $user;
 
+  // Adjust language based on location
+  $user_country_code = dosomething_settings_get_geo_country_code();
+  $language = dosomething_global_convert_country_to_language($user_country_code);
+  user_save($account, array('language' => $language));
```

Comment:
For what it's worth, `array()` vs `[]` is "syntactic sugar"
http://commonphp.blogspot.com/2011/08/php-54-skin-deep-syntactic-sugar.html
I think it's a question of "taste" and "style" but perhaps I'm missing something?!?